dll/win32/browseui/internettoolbar.cpp

```diff
@@ -1928,3 +1917,14 @@ LRESULT CInternetToolbar::OnWinIniChange(UINT uMsg, WPARAM wParam, LPARAM lParam
 
     return lres;
 }
+
+LRESULT CInternetToolbar::OnBrowseUISettingChanged(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    if (!pSettings)
+        pSettings = (BrowseUISettings*)lParam;
```

Comment:
Note, that ideally it would be nice to have the internettoolbar's pSettings pointer initialized to a valid non-NULL value as soon as the internettoolbar object gets created, which should be, in principle, done when a CShellBrowser window gets created too.

Reply:
I agree, but I don't think it's possible given the architecture of our browseui. This is why I was so against moving the settings object to shell browser. The internettoolbar object is created using `CInternetToolbar_CreateInstance` and we only have access to its window handle.

Reply:
Should I resolve this now? pSettings now gets set during `SiteSet()`